### PR TITLE
feature: pouch restful api url support version param

### PR DESCRIPTION
**Pull request by HusterWan**
Signed-off-by: Michael Wan <zirenwan@gmail.com>

### Ⅰ. Describe what this PR did
pouch restful api url support version param

### Ⅳ. Describe how to verify it
execute cli command
```
root@osboxes:pouch (zr/support-version *%) -> pouch ps
Name     ID       Status      Created      Image                                            Runtime
test-2   c84122   Up 3 days   3 days ago   registry.hub.docker.com/library/busybox:latest   runc
```

pouchd logs
```
DEBU[2018-03-12 01:59:18.674256594] Calling GET /v0.2.0/images/json, client @
```

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -17,6 +17,8 @@ import (
 var (
 	defaultHost    = "unix:///var/run/pouchd.sock"
 	defaultTimeout = time.Second * 10
+	// defaultVersion is the version of the current stable API
+	defaultVersion = "0.2.0"
```
Comment (allencloud) [inline client/client.go]: This version should be compatible with Moby, so it should be v1.24, I think.

File: apis/server/router.go [hunk 2]
```diff
@@ -17,50 +17,53 @@ import (
...
 
 	// network
-	r.Path("/networks").Methods(http.MethodGet).Handler(s.filter(s.listNetwork))
-	r.Path("/networks/create").Methods(http.MethodPost).Handler(s.filter(s.createNetwork))
-	r.Path("/networks/{name:.*}").Methods(http.MethodGet).Handler(s.filter(s.getNetwork))
-	r.Path("/networks/{name:.*}").Methods(http.MethodDelete).Handler(s.filter(s.deleteNetwork))
+	r.Path(versionMatcher + "/networks").Methods(http.MethodGet).Handler(s.filter(s.listNetwork))
+	r.Path(versionMatcher + "/networks/create").Methods(http.MethodPost).Handler(s.filter(s.createNetwork))
+	r.Path(versionMatcher + "/networks/{name:.*}").Methods(http.MethodGet).Handler(s.filter(s.getNetwork))
+	r.Path(versionMatcher + "/networks/{name:.*}").Methods(http.MethodDelete).Handler(s.filter(s.deleteNetwork))
```
Comment (allencloud) [inline apis/server/router.go]: I do not think the above code is reasonable, so much redundant code. I will try to make it better.

Comment (HusterWan) [reply apis/server/router.go]: Yes, we may need a router wapper.

File: client/client.go (hunk 1 above)
Comment (HusterWan) [reply client/client.go]: agree

**Comment by allencloud** [comment]
Please add some API integration test for this PR. Thanks a lot. @HusterWan

**Comment by allencloud** [comment]
LGTM

**Pull request merged**